lib/AST/DiagnosticEngine.cpp

```diff
@@ -585,10 +585,12 @@ static void formatDiagnosticArgument(StringRef Modifier,
   }
   case DiagnosticArgumentKind::TypeRepr:
     assert(Modifier.empty() && "Improper modifier for TypeRepr argument");
+    assert(Arg.getAsTypeRepr() && "TypeRepr argument has no name");
```

Comment:
Since the argument is a pointer, I think `TypeRepr argument is null` would be more accurate.

Reply:
Updated.